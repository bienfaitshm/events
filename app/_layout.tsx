import FontAwesome from "@expo/vector-icons/FontAwesome";
import { useFonts } from "expo-font";
import { SplashScreen, Stack } from "expo-router";
import { useEffect } from "react";
import ThemeProvider from "../providers/ThemeProvider";
import AuthenticationProvider from "../providers/AuthenticationProvider";
import ApiClientProvider from "../providers/ApiClientProvider";

export const unstable_settings = {
    initialRouteName: "(auth)/index",
};

export default function _Layout() {
    const [loaded, error] = useFonts({
        SpaceMono: require("../assets/fonts/SpaceMono-Regular.ttf"),
        ...FontAwesome.font,
    });

    // Expo Router uses Error Boundaries to catch errors in the navigation tree.
    useEffect(() => {
        if (error) throw error;
    }, [error]);

    return (
        <>
            {/* Keep the splash screen open until the assets have loaded. In the future, we should just support async font loading with a native version of font-display. */}
            {!loaded && <SplashScreen />}
            {loaded && <RootLayoutNav />}
        </>
    );
}

export function RootLayoutNav() {
    return (
        <>
            <ThemeProvider>
                <ApiClientProvider>
                    <AuthenticationProvider>
                        <Stack
                            screenOptions={{
                                contentStyle: {
                                    backgroundColor: "white",
                                },
                            }}
                        ></Stack>
                    </AuthenticationProvider>
                </ApiClientProvider>
            </ThemeProvider>
        </>
    );
}
