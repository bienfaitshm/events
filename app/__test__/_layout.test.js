import * as React from 'react';
import renderer from 'react-test-renderer';

import {default as Layout ,RootLayoutNav } from '../_layout';

it(`renders correctly Layout`, () => {
  const tree = renderer.create(<Layout />).toJSON();
  expect(tree).toMatchSnapshot();
});

it(`renders correctly RootLayoutNav`, () => {
  const tree = renderer.create(<RootLayoutNav />).toJSON();
  expect(tree).toMatchSnapshot();
});

