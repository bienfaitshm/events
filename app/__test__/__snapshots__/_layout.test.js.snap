// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Layout 1`] = `null`;

exports[`renders correctly RootLayoutNav 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
